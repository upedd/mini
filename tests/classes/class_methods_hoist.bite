class Foo {
    add_to_10(a) {
        return this.add(a, this.get_10());
    }

    get10() {
        return 10;
    }

    add(a, b) {
        return a + b;
    }
}
native print;

let instance = Foo();

print(instance.get10());
print(instance.add(3, 5));
print(instance.add_to_10(7));