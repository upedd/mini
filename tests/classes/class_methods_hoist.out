10
8
17